Helper/GraphQL/Client.php

```diff
@@ -0,0 +1,194 @@
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\HTTP\ZendClientFactory;
+use Bolt\Boltpay\Model\ResponseFactory;
+use Bolt\Boltpay\Model\RequestFactory;
+use Bolt\Boltpay\Helper\Log as LogHelper;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\Phrase;
+use Zend_Http_Client_Exception;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Shared\Utils;
+
+
+/**
+ * Boltpay GraphQLAPI helper
+ *
+ * @SuppressWarnings(PHPMD.TooManyFields)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class Client extends AbstractHelper
+{
+
+    /**
+     * MerchantAPI graphQL endpoint
+     */
+    const MERCHANT_API_GQL_ENDPOINT = 'v2/merchant/api';
+
+    /**
+     * @var ZendClientFactory
+     */
+    private $httpClientFactory;
+
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+    /**
+     * Response factory
+     *
+     * @var ResponseFactory
+     */
+    private $responseFactory;
+
+    /**
+     * Request factory
+     *
+     * @var RequestFactory
+     */
+    private $requestFactory;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @param Context $context
+     * @param ZendClientFactory $httpClientFactory
+     * @param ConfigHelper $configHelper
+     * @param ResponseFactory $responseFactory
+     * @param RequestFactory $requestFactory
+     * @param LogHelper $logHelper
+     * @param Bugsnag $bugsnag
+     * @codeCoverageIgnore
+     */
+    public function __construct(
+        Context $context,
+        ZendClientFactory $httpClientFactory,
+        ConfigHelper $configHelper,
+        ResponseFactory $responseFactory,
+        RequestFactory $requestFactory,
+        LogHelper $logHelper,
+        Bugsnag $bugsnag
+    ) {
+        parent::__construct($context);
+        $this->httpClientFactory = $httpClientFactory;
+        $this->configHelper = $configHelper;
+        $this->responseFactory = $responseFactory;
+        $this->requestFactory = $requestFactory;
+        $this->logHelper = $logHelper;
+        $this->bugsnag = $bugsnag;
+    }
+
+    private function makeGQLCall($query, $operation, $variables) {
+        $result = $this->responseFactory->create();
+        $client = $this->httpClientFactory->create();
+        error_log("KOKOKOKOK");
+
+        $apiKey = $this->configHelper->getApiKey();
+
+        $gql_request = array(
+            "operationName" => $operation,
+            "variables" => $variables,
+            "query" => $query
+        );
+
+        $requestData = json_encode($gql_request, JSON_UNESCAPED_SLASHES);
+
+        $apiURL = $this->configHelper->getApiUrl() . self::MERCHANT_API_GQL_ENDPOINT;
+        error_log($apiURL);
+
+        $client->setUri($apiURL);
+
+        $client->setConfig(['maxredirects' => 0, 'timeout' => 30]);
+
+        $headers =  [
```

Comment:
Hm, can we somehow share this with `Helper/Api`?

Reply:
yeah let me move to a constants class